File: backend/Projetos/KeySystems.ERP.ContasAPagar/src/KeySystems.ERP.ContasAPagar.Core/DomainModel/ContasAPagar/Titulo.cs
using System;
using System.Collections.Generic;

namespace KeySystems.ERP.ContasAPagar.Core.DomainModel.ContasAPagar
{
    public class Titulo : IEntity<int>
    {
        public Titulo(int idEmpresa, int idTipoTitulo, int idFornecedor)
        {
            if (idEmpresa == default(int))
                throw new ArgumentException("Titulo", "Empresa não informada");

            if (idTipoTitulo == default(int))
                throw new ArgumentException("Titulo", "Tipo Titulo não informado");

            if (idFornecedor == default(int))
                throw new ArgumentException("Titulo", "Fornecedor não informado");

            this.IdEmpresa = idEmpresa;
            this.IdTipoTitulo = idTipoTitulo;
            this.IdFornecedor = idFornecedor;
        }

        public int Id { get; private set; }
        public int IdEmpresa { get; private set; }
        public int IdTipoTitulo { get; private set; }
        public int IdFornecedor { get; private set; }
        public decimal Total { get; set; }

        public List<Parcela> Parcelas { get; private set; } = new List<Parcela>();

        

    }
}

File: frontend/src/app/modules/contas-pagar/titulo/shared/models/titulo.ts

import { TipoTitulo } from "./tipoTitulo";
import {TipoPgto } from './tipoPgto';
import {Parcela } from './parcelas';


export class Titulo {
    
    constructor(tipoTitulo:TipoTitulo, tipoPgto: TipoPgto) {
        this.tipoTitulo = tipoTitulo;
        this.tipoPgto = tipoPgto;
    }

    private _id : number;
    public get id() : number {
        return this._id;
    }
    public set id(v : number) {
        this._id = v;
    }
    
    private _tipoTitulo : TipoTitulo;
    public get tipoTitulo() : TipoTitulo {
        return this._tipoTitulo;
    }
    public set tipoTitulo(v : TipoTitulo) {
        this._tipoTitulo = v;
    }
    
    private _tipoPgto : TipoPgto;
    public get tipoPgto() : TipoPgto {
        return this._tipoPgto;
    }
    public set tipoPgto(v : TipoPgto) {
        this._tipoPgto = v;
    }
    
    private _parcelas : Array<Parcela>;
    public get parcelas() : Array<Parcela> {
        return this._parcelas;
    }
    public set parcelas(v : Array<Parcela>) {
        this._parcelas = v;
    }
    
}
